[docs] EAS Builds: add 'Setup' page & more (#9754)
expo/expo: docs/pages/build/eas-json.md

```

- `"workflow": "generic"` indicates that your project is a generic one.

> - `credentialsSource` defines the credentials source for the project build. If you want to take advantage of your own `credentials.json` file, set it to `local` ([learn more on this here](../advanced-credentials-configuration/)). If you always want to use credentials stored on Expo servers, choose `remote`. If you're not sure what to do but you probably won't be needing to run builds from a CI, choose `auto` (this is the default option).

Reviewer: ```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to take advantage of your own `credentials.json` file, set this to `local` ([learn more on this here](../advanced-credentials-configuration/)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do but you probably won't be running builds from a CI, choose `auto` (this is the default option).
```

```

- `"workflow": "generic"` indicates that your project is a generic one.

> - `credentialsSource` defines the credentials source for the project build. If you want to take advantage of your own `credentials.json` file, set it to `local` ([learn more on this here](../advanced-credentials-configuration/)). If you always want to use credentials stored on Expo servers, choose `remote`. If you're not sure what to do but you probably won't be needing to run builds from a CI, choose `auto` (this is the default option).

Reviewer: ```suggestion
- `credentialsSource` defines the source of credentials for this build profile. If you want to take advantage of your own `credentials.json` file, set this to `local` ([learn more on this here](../advanced-credentials-configuration/)). If you want to use the credentials Expo already has stored for you, choose `remote`. If you're not sure what to do but you probably won't be running builds from a CI, choose `auto` (this is the default option).
```

}
```

> If you'd like to build your iOS project with your custom `Gymfile` ([learn more on this here](../ios-builds/#building-ios-projects-with-fastlane)) and you've defined a different output directory than `ios/build` use the following example:

Reviewer: ```suggestion
If you'd like to build your iOS project with a custom `Gymfile` ([learn more on this here](../ios-builds/#building-ios-projects-with-fastlane)) where you've defined a different output directory than `ios/build`, use the following example:
```
